source/dub/project.d

```diff
@@ -230,10 +230,15 @@ class Project {
 
 		This will emit warnings to `stderr` if any discouraged names or
 		dependency patterns are found.
+
+		Returns: 
 	*/
-	void validate()
+	ValidationResult validate()
 	{
-		// some basic package lint
+		ValidationResult validity;
+
+		// some basic package name lint
+		validity.validName = true;
```

Comment:
Struct Initialization here with all true in the beginning here?

Reply:
I wanted the init state to be invalid, but I think I can make a fancy tupleof foreach here